Generals/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5428,21 +5428,33 @@ void InGameUI::selectNextIdleWorker( void )
 	else
 	{
 		Drawable *selectedDrawable = TheInGameUI->getFirstSelectedDrawable();	
+		ObjectList uniqueIdleWorkers;
+
+		// Find unique selectables to avoid selecting the same or a previous container if multiple idle workers are contained
```

Comment:
I think this approach is technically correct and the least error prone.

The only upset with this is that it needs to do these allocations, then populate this list, on every idle worker button press. I did have a try at building the idle worker data structures to have container separation already organized on addIdleWorker, but this is very tricky to do because of containers inside containers (Worker inside Humvee inside Chinook) and adds a great level of complexity to this.

Reply:
I did consider other possible more efficient solutions but as you may have figured, containers can change while the worker remains idle which significantly complicates matters. I don't think efficiency is really much of a concern here seeing as it is a seldom-used action that is not typically spammed nor frame critical. I believe the simplicity of the solution at hand and its isolation to a single block of a dozen or so lines is preferable to adding significant complexity for negligible performance gain.

Reply:
Yes I agree. The simplicity and correctness of the implementation is way more important here.